### ISSUE TITLE
dotnet amazon linux builds broken

### ISSUE THREAD

**Issue by ptdel** [dotnet/core#2423]
```Processing files: openssl-libs-1.0.0-0.x86_64
Provides: openssl-libs = 1.0.0-0 openssl-libs(x86-64) = 1.0.0-0
Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1
Checking for unpackaged file(s): /usr/lib/rpm/check-files /usr/src/rpm/BUILDROOT/openssl-libs-1.0.0-0.x86_64
Wrote: /usr/src/rpm/RPMS/x86_64/openssl-libs-1.0.0-0.x86_64.rpm
Executing(%clean): /bin/sh -e /var/tmp/rpm-tmp.13DnAm
```
for whatever reason this chose today to start hanging forever during deployments.  I don't see an actual %clean step in the rpm spec.

_Originally posted by @ptdel in https://github.com/dotnet/core/issues/930#issuecomment-470206637_

**Comment by leecow** [comment]
Are you trying to install the update which released today? Are you using the rpms from our feed (packages.microsoft.com) or something else?

@dagood

**Comment by leecow** [comment]
Probably some environment differences so, grain of salt, I was able to install and run basic tests with the new 2.1 and 2.2 SDKs on a clean Amazon Linux Docker image.

```
bash-4.2# dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.2.105
 Commit:    7cecb35b92

Runtime Environment:
 OS Name:     amzn
 OS Version:  2
 OS Platform: Linux
 RID:         linux-x64
 Base Path:   /usr/share/dotnet/sdk/2.2.105/

Host (useful for support):
  Version: 2.2.3
  Commit:  6b8ad509b6

.NET Core SDKs installed:
  2.1.505 [/usr/share/dotnet/sdk]
  2.2.105 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.9 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.9 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.9 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
bash-4.2# cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
```

**Comment by dagood** [comment]
@wfurt, it sounds like this is specific to your spec file in https://github.com/dotnet/core/issues/930#issuecomment-404296184?

**Comment by wfurt** [comment]
That spec file only creates fake dependencies so our package would install without much wrestling. From what distro did you get RPM @leecow ? I did quick check with Centos7 and it still asks for openssl-libs (missing on AWS)

```
[root@cent tmp]# rpm -qRp dotnet-runtime-deps-latest-centos.7-x64.rpm
openssl-libs
libicu
krb5-libs
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(CompressedFileNames) <= 3.0.4-1
```

note for @ptdel that the fake spec does not build anything. If it hangs you will need to debug it on your system.  Make sure you don't have other processes holding rpm database lock.

**Comment by leecow** [comment]
https://packages.microsoft.com/rhel/7/prod/

**Comment by wfurt** [comment]
what do you get from: `rpm -qi openssl-libs`. Maybe there is standard provider now. 
I don't have access to any AMI Linux to check.

**Comment by ptdel** [comment]
@wfurt doesn't appear to be any official packages via yum or rpm for openssl-libs still.  not able to see anything under /var/lib/rpm that would cause issues like .rpm.lock or .dbenv.lock. @leecow I've been grabbing it from https://go.microsoft.com/fwlink/?LinkID=835019 I'll try re-running bootstraps fetching dotnet sdk from the rpm and see if that helps.  This is for Jenkins workers I might just need to own up to making some windows nodes

**Comment by wfurt** [comment]
BTW you should be also able to use rpm -i --nodeps --force dotnet-runtime-deps*.rpm.
With that, rest should install as well.

**Comment by ptdel** [comment]
@wfurt would this be to just skip openssl-libs during the dotnet runtime install?  I gave it a go, but it appears the lack of openssl-libs is a show stopper:
```
--> Processing Dependency: openssl-libs for package: dotnet-runtime-deps-2.2-2.2.3-1.x86_64

Packages skipped because of dependency problems:
    dotnet-runtime-deps-2.2-2.2.3-1.x86_64 from /dotnet-runtime-deps-2.2.3-rhel.7-x64Checking for unpackaged file(s): /usr/lib/rpm/check-files /usr/src/rpm/BUILDROOT/openssl-libs-1.0.0-0.x86_64
```  
I switched rpmbuild's flags from `--bb` to `--bi` thinking maybe I could skip the clean step, but its just stopping at the previous step of 
```
Checking for unpackaged file(s): /usr/lib/rpm/check-files /usr/src/rpm/BUILDROOT/openssl-libs-1.0.0-0.x86_64
``` 
looking again under the rpm directories I still don't see anything to indicate that yum or rpm have any kind of lock on anything.

If I'm grabbing the dotnet runtime, and sdk from packages.microsoft.com, and using this psuedo-rpm spec is there anything atypical in this setup?
```
curl -SL -o openssl-libs.ami.spec.txt "https://github.com/dotnet/core/files/2186067/openssl-libs-ami.spec.txt"
rpmbuild --bi openssl-libs.ami.spec.txt
rpm -i openssl-libs-1.0.0-0.x86_64.rpm

DOTNET_RUNTIME=https://packages.microsoft.com/rhel/7/prod/dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm
curl -SL -o dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm $DOTNET_RUNTIME
yum -y install dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm

DOTNET=https://packages.microsoft.com/rhel/7/prod/dotnet-sdk-2.2.105-x64.rpm
curl -SL -o dotnet-sdk-2.2.105-x64.rpm $DOTNET
yum -y install dotnet-sdk-2.2.105-x64.rpm
```
I'll admit I'm not very versed in the .NET ecosystem so maybe I'm leaving out entire dependencies or something of the sort

**Comment by karelz** [comment]
@wfurt @tmds @RheaAyase can you check what is left here?

**Comment by wfurt** [comment]
I lost track @karelz. 
```
$ rpm -qpR https://packages.microsoft.com/rhel/7/prod/dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm
warning: https://packages.microsoft.com/rhel/7/prod/dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm: Header V4 RSA/SHA256 Signature, key ID be1229cf: NOKEY
openssl-libs
libicu
krb5-libs
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(CompressedFileNames) <= 3.0.4-1
```

Since the SDK does not depend on openssl-libs, one could simple do `yum install   libicu krb5-libs` to get real dependencies and than use `rpm -i --nodep runtime-deps.rpm` 

After that, rest should just work. 
I don't have access to amazon linux so it is hard for me to test. I don't know if the docker image @leecow used is same as real AWS service.

**Comment by tmds** [comment]
I'm not familiar with Amazon Linux either, afaik it is based on CentOS7.
@ptdel a Dockerfile that reproduces the issue would be helpful.

@omajid is our expert for packaging. @normj (from Amazon) may be able to help too.

**Comment by leecow** [comment]
This is what I used and don't know if it's the same as what's available on AWS.

```
$docker search amazon
NAME                                   DESCRIPTION                                     STARS               OFFICIAL            AUTOMATED
amazonlinux                            Amazon Linux provides a stable, secure, and …   636                 [OK]
```

**Comment by ptdel** [comment]
@tmds @leecow I'm not sure the docker image is a totally accurate representation of the linux AMI, but I'd have to check.

We had to move on from this unfortunately due to the daily grind.  When it was working, the environment was tenuous, not necessarily because of the sdk, but the tools using it.  For example the Jenkins Powershell plugin _technically_ works on linux or windows, but I had to fenagle it to actually use powershell on linux in a major way.  The overhead wasn't worth it ultimately given that this is for our CICD.  Ultimately we decided that while the prospect of using dotnet and powershell on Linux is appealing, tooling out there we need hasn't warmed up to that idea yet.  I really would like to take another shot at this at some point though, as the prospect of building for windows on linux would at least dole out some street creds.

@wfurt I've attempted to use the prescribed methods with RPM and yum, but no luck.
```
[ec2-user@ip-10-0-15-192 ~]$ sudo rpm -i --nodeps dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm
warning: dotnet-runtime-deps-2.2.3-rhel.7-x64.rpm: Header V4 RSA/SHA256 Signature, key ID be1229cf: NOKEY
        package dotnet-runtime-deps-2.2-2.2.3-1.x86_64 is already installed
```
I can confirm that I've installed it via this method previously

```
[ec2-user@ip-10-0-15-192 ~]$ sudo yum install dotnet-sdk-2.2.105-x64.rpm
Loaded plugins: priorities, update-motd, upgrade-helper
Examining dotnet-sdk-2.2.105-x64.rpm: dotnet-sdk-2.2-2.2.105-1.x86_64
Marking dotnet-sdk-2.2.105-x64.rpm to be installed
Resolving Dependencies
amzn-main/latest                                                                                | 2.1 kB  00:00:00
amzn-updates/latest                                                                             | 2.5 kB  00:00:00
--> Running transaction check
---> Package dotnet-sdk-2.2.x86_64 0:2.2.105-1 will be installed
--> Processing Dependency: dotnet-runtime-2.2 >= 2.2.3 for package: dotnet-sdk-2.2-2.2.105-1.x86_64
--> Processing Dependency: aspnetcore-runtime-2.2 >= 2.2.3 for package: dotnet-sdk-2.2-2.2.105-1.x86_64
--> Finished Dependency Resolution
Error: Package: dotnet-sdk-2.2-2.2.105-1.x86_64 (/dotnet-sdk-2.2.105-x64)
           Requires: dotnet-runtime-2.2 >= 2.2.3
Error: Package: dotnet-sdk-2.2-2.2.105-1.x86_64 (/dotnet-sdk-2.2.105-x64)
           Requires: aspnetcore-runtime-2.2 >= 2.2.3
 You could try using --skip-broken to work around the problem
** Found 1 pre-existing rpmdb problem(s), 'yum check' output follows:
dotnet-runtime-deps-2.2-2.2.3-1.x86_64 has missing requires of openssl-libs
```
we have the dependencies but it seems like funkier things are happening.  I'm also not sure why the installed package from upstream goes from `2.2.3` to `2.2-2.2.3-1`

From the uninformed perspective, I have to ask what the need for a fake openssl-lib dependency is?  I remember seeing some reason for it listed somewhere, is there something about using the actual provided openssl libs on linux that is prohibitive?

**Comment by leecow** [comment]
@bartonjs for possible info on the openssl-lib dependency.

**Comment by bartonjs** [comment]
Our crypto stack depends on OpenSSL provided by the system.  From what I can see, for CentOS 7 the package that provides the OpenSSL libraries is [openssl-libs](https://centos.pkgs.org/7/centos-x86_64/openssl-libs-1.0.2k-16.el7.x86_64.rpm.html).  (I chose CentOS 7 because of a previous comment suggesting that's what Amazon Linux is based on.)

If Amazon Linux gets libcrypto.so.{1.1, 1.0.2, 1.0.0, 10} (whichever one it uses) via some other package, then presumably we need to register a different package dependency for that distro.

**Comment by ptdel** [comment]
```Linux ip-10-0-15-192 4.14.77-69.57.amzn1.x86_64 #1 SMP Tue Nov 6 21:32:55 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
[ec2-user@ip-10-0-15-192 ~]$ repoquery -l openssl | grep /usr/lib/
/usr/lib/.libcrypto.so.1.0.2k.hmac
/usr/lib/.libcrypto.so.10.hmac
/usr/lib/.libssl.so.1.0.2k.hmac
/usr/lib/.libssl.so.10.hmac
/usr/lib/libcrypto.so.1.0.2k
/usr/lib/libcrypto.so.10
/usr/lib/libssl.so.1.0.2k
/usr/lib/libssl.so.10
/usr/lib/openssl
/usr/lib/openssl/engines
/usr/lib/openssl/engines/lib4758cca.so
/usr/lib/openssl/engines/libaep.so
/usr/lib/openssl/engines/libatalla.so
/usr/lib/openssl/engines/libcapi.so
/usr/lib/openssl/engines/libchil.so
/usr/lib/openssl/engines/libcswift.so
/usr/lib/openssl/engines/libgmp.so
/usr/lib/openssl/engines/libnuron.so
/usr/lib/openssl/engines/libpadlock.so
/usr/lib/openssl/engines/libsureware.so
/usr/lib/openssl/engines/libubsec.so
```
This is typically the most widely used image of Amazon Linux https://aws.amazon.com/amazon-linux-ami/2018-03-packages/ , but there is a newer revision https://aws.amazon.com/amazon-linux-2/faqs/

**Comment by wfurt** [comment]
The original issue I was trying to solve with fake package @bartonjs was lack of openssl-libs package. There was openssl on the system but the rpm would not install as it would see missing dependency. This was unrelated to actual crypto.

**Comment by wfurt** [comment]
I just realized I missed part of your response @ptdel 
Yes, it is unfortunate to go through the extra steps. But Amazon linux is not on supported list: https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2-supported-os.md so we do not run tests on it or build for it. That it makes it community supported. You can also check with distro vendor e.g.  Amazon to make packages for their Linux flavor.

**Comment by LatentLag** [comment]
I've been dealing with a similar issue on CentOS 7. I've been trying to run dotnet core app against runtime 2.1 or 2.2 (netcoreapp2.1 or netcoreapp2.2) but running into the issue. It seems that the runtime was built against openssl-libs 1.1 but CentOS 7 only has openssl-libs version 1.0.2k available.

Since it was mentioned the "SDK" doesn't have a dependency on openssl I'm trying to install that.

Here's the dotnet core error:
`$ LD_DEBUG=libs dotnet my-application.dll`
``` (trimmed excessive logging)
      2304:	calling init: /lib64/libssl.so.10
      2304:	
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: DSA_get0_key (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: DSA_get0_pqg (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: DSA_get_method (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: DSA_set0_key (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: DSA_set0_pqg (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EVP_CIPHER_CTX_reset (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EVP_PKEY_up_ref (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: HMAC_CTX_free (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: HMAC_CTX_new (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: OpenSSL_version (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_get0_crt_params (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_get0_factors (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_get0_key (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_meth_get_flags (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_set0_crt_params (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_set0_factors (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: RSA_set0_key (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: SSL_is_init_finished (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: SSL_CTX_set_options (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: SSL_CTX_set_security_level (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: SSL_session_reused (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_CRL_get0_nextUpdate (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_get_version (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_get_X509_PUBKEY (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_get0_notBefore (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_get0_notAfter (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_get0_tbs_sigalg (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_NAME_get0_der (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_STORE_CTX_get0_cert (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_STORE_CTX_get0_chain (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_STORE_CTX_get0_untrusted (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: X509_up_ref (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EC_GF2m_simple_method (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EC_GROUP_get_curve_GF2m (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EC_GROUP_set_curve_GF2m (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EC_POINT_get_affine_coordinates_GF2m (fatal)
      2304:	/lib64/libssl.so.10: error: symbol lookup error: undefined symbol: EC_POINT_set_affine_coordinates_GF2m (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/System.Security.Cryptography.Native.OpenSsl.so: error: symbol lookup error: undefined symbol: DllMain (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_EndsWithW (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_IndexOfW (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_GetLocaleInfoIntW (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_GetLocaleInfoGroupingSizesW (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_CoTaskMemAlloc (fatal)
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_LocalAlloc (fatal)
      2304:	find library=__Internal.so [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/__Internal.so
      2304:	  trying file=/lib64/__Internal.so
      2304:	  trying file=/usr/lib64/tls/__Internal.so
      2304:	  trying file=/usr/lib64/__Internal.so
      2304:	
      2304:	find library=lib__Internal.so [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/lib__Internal.so
      2304:	  trying file=/lib64/lib__Internal.so
      2304:	  trying file=/usr/lib64/tls/lib__Internal.so
      2304:	  trying file=/usr/lib64/lib__Internal.so
      2304:	
      2304:	find library=__Internal [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/__Internal
      2304:	  trying file=/lib64/__Internal
      2304:	  trying file=/usr/lib64/tls/__Internal
      2304:	  trying file=/usr/lib64/__Internal
      2304:	
      2304:	find library=lib__Internal [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/lib__Internal
      2304:	  trying file=/lib64/lib__Internal
      2304:	  trying file=/usr/lib64/tls/lib__Internal
      2304:	  trying file=/usr/lib64/lib__Internal
      2304:	

Unhandled Exception:       2304:	find library=Microsoft.DiaSymReader.Native.amd64.dll [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	
      2304:	find library=Microsoft.DiaSymReader.Native.amd64.dll [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	
      2304:	find library=Microsoft.DiaSymReader.Native.amd64.dll [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	
      2304:	find library=Microsoft.DiaSymReader.Native.amd64.dll [0]; searching
      2304:	 search cache=/etc/ld.so.cache
      2304:	 search path=/lib64/tls:/lib64:/usr/lib64/tls:/usr/lib64		(system search path)
      2304:	  trying file=/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/tls/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	  trying file=/usr/lib64/Microsoft.DiaSymReader.Native.amd64.dll
      2304:	
      2304:	/usr/share/dotnet/shared/Microsoft.NETCore.App/2.1.11/libcoreclr.so: error: symbol lookup error: undefined symbol: PAL_LocalFree (fatal)
(trimmed excessive logging)
```

Environment information:
```
$ yum info openssl-libs
Loaded plugins: fastestmirror, versionlock
Loading mirror speeds from cached hostfile
 * base: centos.host-engine.com
 * epel: fedora-epel.mirror.lstn.net
 * extras: centos.mirror.constant.com
 * updates: centos.s.uw.edu
Excluding 8 updates due to versionlock (use "yum versionlock status" to show them)
Installed Packages
Name        : openssl-libs
Arch        : x86_64
Epoch       : 1
Version     : 1.0.2k
Release     : 16.el7
Size        : 3.1 M
Repo        : installed
Summary     : A general purpose cryptography library with TLS implementation
URL         : http://www.openssl.org/
License     : OpenSSL
Description : OpenSSL is a toolkit for supporting cryptography. The openssl-libs
            : package contains the libraries that are used by various applications which
            : support cryptographic algorithms and protocols.

Available Packages
Name        : openssl-libs
Arch        : i686
Epoch       : 1
Version     : 1.0.2k
Release     : 16.el7_6.1
Size        : 994 k
Repo        : updates/7/x86_64
Summary     : A general purpose cryptography library with TLS implementation
URL         : http://www.openssl.org/
License     : OpenSSL
Description : OpenSSL is a toolkit for supporting cryptography. The openssl-libs
            : package contains the libraries that are used by various applications which
            : support cryptographic algorithms and protocols.

Name        : openssl-libs
Arch        : x86_64
Epoch       : 1
Version     : 1.0.2k
Release     : 16.el7_6.1
Size        : 1.2 M
Repo        : updates/7/x86_64
Summary     : A general purpose cryptography library with TLS implementation
URL         : http://www.openssl.org/
License     : OpenSSL
Description : OpenSSL is a toolkit for supporting cryptography. The openssl-libs
            : package contains the libraries that are used by various applications which
            : support cryptographic algorithms and protocols.

$ dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   2.1.700
 Commit:    c2ef055a0f

Runtime Environment:
 OS Name:     centos
 OS Version:  7
 OS Platform: Linux
 RID:         centos.7-x64
 Base Path:   /usr/share/dotnet/sdk/2.1.700/

Host (useful for support):
  Version: 2.1.11
  Commit:  d6a5616240

.NET Core SDKs installed:
  2.1.700 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

$ cat /etc/centos-release
CentOS Linux release 7.6.1810 (Core)
```

**Comment by wfurt** [comment]
That is different issue @LatentLag. Dotnet has portable shim to work with multiple versions of openssl. (and 1.0.2 is good enough) You may see probes for various functions as part of the process of what is available. Do you have any functional problem? If so, open separate issue.

**Comment by carlossanlop** [comment]
No comments since June, so closing. @ptdel please open a new issue if you keep seeing problems.
